server/src/internalClusterTest/java/org/elasticsearch/action/support/master/TransportMasterNodeActionIT.java

```diff
@@ -107,27 +115,98 @@ public void onFailure(Exception e) {
                     }
                 });
 
-            safeAwait(electionWonLatch);
+            // Wait until the old master has acknowledged the new master's election
+            safeAwait(previousMasterKnowsNewMasterIsElectedLatch);
+            logger.info("New master is elected");
```

Comment:
because we're preventing cluster state updates being applied on `newMaster`, we watch the previous master's cluster state to determine when we've won the election. We need to wait for the previous master to apply the term bump anyway before we send the to-be-redirected request so this serves multiple purposes.